packages/expo-updates/ios/EXUpdates/EXUpdatesAppController.m

```diff
@@ -307,6 +323,103 @@ - (void)runReaper
+  [remoteAppLoader loadUpdateFromUrl:_config.updateUrl onManifest:^BOOL(EXUpdatesUpdate *update) {
+    return [self->_selectionPolicy shouldLoadNewUpdate:update withLaunchedUpdate:self.launchedUpdate filters:update.manifestFilters];
+  } asset:^(EXUpdatesAsset *asset, NSUInteger successfulAssetCount, NSUInteger failedAssetCount, NSUInteger totalAssetCount) {
+    // do nothing for now
+  } success:^(EXUpdatesUpdate * _Nullable update) {
+    self->_remoteLoadStatus = update ? EXUpdatesRemoteLoadStatusNewUpdateLoaded : EXUpdatesRemoteLoadStatusIdle;
+    [self->_errorRecovery notifyNewRemoteLoadStatus:self->_remoteLoadStatus];
+  } error:^(NSError *error) {
+    self->_remoteLoadStatus = EXUpdatesRemoteLoadStatusIdle;
+    [self->_errorRecovery notifyNewRemoteLoadStatus:self->_remoteLoadStatus];
+  }];
+}
+
+- (void)markFailedLaunchForLaunchedUpdate
+{
+  dispatch_async(_database.databaseQueue, ^{
+    EXUpdatesUpdate *launchedUpdate = self.launchedUpdate;
+    if (!launchedUpdate) {
+      return;
+    }
+    NSError *error;
+    [self->_database incrementFailedLaunchCountForUpdate:launchedUpdate error:&error];
+    if (error) {
+      NSLog(@"Unable to mark update as failed in the local DB: %@", error.localizedDescription);
+    }
+  });
+}
+
+- (void)markSuccessfulLaunchForLaunchedUpdate
+{
+  dispatch_async(_database.databaseQueue, ^{
+    EXUpdatesUpdate *launchedUpdate = self.launchedUpdate;
+    if (!launchedUpdate) {
+      return;
+    }
+    NSError *error;
+    [self->_database incrementSuccessfulLaunchCountForUpdate:launchedUpdate error:&error];
+    if (error) {
+      NSLog(@"Failed to increment successful launch count for update: %@", error.localizedDescription);
+      return;
+    }
+
+    NSError *selectAllUpdatesError;
+    NSArray<EXUpdatesUpdate *> *allUpdates = [self->_database allUpdatesWithConfig:self->_config error:&selectAllUpdatesError];
+    if (selectAllUpdatesError) {
+      NSLog(@"Failed to select updates to mark as outdated: %@", error.localizedDescription);
+      return;
+    }
+
+    NSError *manifestFiltersError;
+    NSDictionary *manifestFilters = [self->_database manifestFiltersWithScopeKey:self->_config.scopeKey error:&manifestFiltersError];
+    if (manifestFiltersError) {
+      NSLog(@"Error selecting manifest filters while marking updates as outdated: %@", error.localizedDescription);
+      return;
+    }
+
+    NSArray<EXUpdatesUpdate *> *outdatedUpdates = [self->_selectionPolicy outdatedUpdatesWithLaunchedUpdate:launchedUpdate updates:allUpdates filters:manifestFilters];
+    NSError *markUpdatesOutdatedError;
+    [self->_database markUpdatesOutdated:outdatedUpdates error:&markUpdatesOutdatedError];
+    if (markUpdatesOutdatedError) {
+      NSLog(@"Failed to mark updates as outdated: %@", error.localizedDescription);
+    }
+  });
```

Comment:
Mentioned this on another PR -- may want to consider writing a query that finds outdated updates instead of marking them as outdated. However that approach might have different behavior if the manifest filters change over time.